redis/src/client.rs

```diff
@@ -56,6 +60,14 @@ impl Client {
         connect(&self.connection_info, None)
     }
 
+    /// Same as [`Self::get_connection`] but with a custom IO Driver.
+    pub fn get_connection_with_io<IO: io::ConnectionDriver>(
+        &self,
+        driver: IO,
```

Comment:
nit: I prefer to use `driver: impl io::ConnectionDriver`

Reply:
Sure thing, will address it once I also understand better your other comments. This way I can handle it together with this one.

Reply:
It's not possible to use `impl` here though, as we also need to specify the same type as RedisResult<GenericConnection<IO>> which AFAIK you cannot make clear if you have 

```
pub fn get_connection_with_io_and_timeout(
        &self,
        driver: impl io::ConnectionDriver,
        timeout: Duration,
    ) -> RedisResult<GenericConnection<impl io::ConnectionDriver>>
```

^ the above works but you lose guarantee that the result type is the same as the input type

Reply:
When I wrote my originalk reply 13 hours ago I hadn't read your `impl` part, I though you wanted to name it `Driver` instead of `IO`, which does make sense if you want that for some reason